pkg/stanza/fileconsumer/reader_factory.go

```diff
@@ -80,6 +83,15 @@ func (b *readerBuilder) withFingerprint(fp *Fingerprint) *readerBuilder {
 	return b
 }
 
+func (b *readerBuilder) withBufferSize() *readerBuilder {
+	if b.readerConfig.fingerprintSize > defaultBufSize {
+		b.bufferSize = b.readerConfig.fingerprintSize
+	} else {
+		b.bufferSize = defaultBufSize
+	}
+	return b
+}
+
```

Comment:
Any reason not to always set the buffer size like this?

Reply:
Now default buffer size is set to 16 kb, default fingerprint size is set 1kb in `filelogreceiver` so the difference is significant, lower buffer for reading could impact speed of reading from the log file (more read operations with smaller buffer), higher fingerprint size could cause higher memory consumption - more data kept in memory to store fingerprint size.

Maybe consequences of setting always  `b.bufferSize = b.readerConfig.fingerprintSize` will not be visible but the issue which I observed exists only for cases when fingerprint size is higher than the buffer.